src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,244 @@
+        workflowBatch <- reifyWorkflowRecords(wfRecs, dataAccess)
+
+        //The submission itself
+        submissionId = wfRecs.head.submissionId
+        submissionRec <- dataAccess.submissionQuery.findById(submissionId).result.map(_.head)
+
+        //The workspace
+        workspaceRec <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result.map(_.head)
+
+        //The workspace's billing project
+        billingProject <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRec.namespace)).map(_.get)
+
+        //The person who submitted the submission, and their token
+        submitter <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId))).map(_.get)
+        token <- DBIO.from(googleServicesDAO.getToken(submitter)).map(_.get)
+
+        //The wdl
+        methodConfig <- dataAccess.methodConfigurationQuery.loadMethodConfigurationById(submissionRec.methodConfigurationId).map(_.get)
+        wdl <- getWdl(methodConfig)
+      } yield {
+
+        val wfOpts = buildWorkflowOpts(workspaceRec, submissionId, submitter, token, billingProject)
+
+        val wfInputsBatch = workflowBatch map { wf =>
+          val methodProps = wf.inputResolutions map {
+            case svv: SubmissionValidationValue if svv.value.isDefined =>
+              svv.inputName -> svv.value.get
+          }
+          MethodConfigResolver.propertiesToWdlInputs(methodProps.toMap)
+        }
+
+        //yield the things we're going to submit to Cromwell
+        (wdl, wfRecs, wfInputsBatch, wfOpts)
+      }
+    }
+
+    import ExecutionJsonSupport.ExecutionServiceWorkflowOptionsFormat
+    val cromwellSubmission = for {
+      (wdl, workflowRecs, wfInputsBatch, wfOpts) <- workflowBatchFuture
+      workflowSubmitResult <- executionServiceDAO.submitWorkflows(wdl, wfInputsBatch, Option(wfOpts.toJson.toString), getUserInfo(credential))
+    } yield {
+      workflowRecs.zip(workflowSubmitResult)
+    }
+
+    cromwellSubmission flatMap { results =>
+      dataSource.inTransaction { dataAccess =>
+        //save successes as submitted workflows and hook up their cromwell ids
+        val successes = results collect {
+          case (wfRec, Left(success: ExecutionServiceStatus)) =>
+            val updatedWfRec = wfRec.copy(externalId = Option(success.id), status = success.status)
+            dataAccess.workflowQuery.updateWorkflowRecord(updatedWfRec)
+        }
+
+        //save error messages into failures and flip them to Failed
+        val failures = results collect {
+          case (wfRec, Right(failure: ExecutionServiceFailure)) =>
+            //TODO: batchify this saveMessages?
+            dataAccess.workflowQuery.saveMessages(Seq(AttributeString(failure.message)), wfRec.id) andThen
+            dataAccess.workflowQuery.updateStatus(wfRec, WorkflowStatuses.Failed)
+        }
+
+        DBIO.seq((successes ++ failures):_*)
+      } map { _ => LookForWorkflows }
```

Comment:
Wouldn't waiting an interval be appropriate here, and returning `ScheduleNextWorkflowQuery` instead?

Currently this goes `LookForWorkflows` <-> `SubmitWorkflowBatch` back and forth without pausing.